DataConnectors/CEF/cef_gather_info.py

```diff
@@ -0,0 +1,152 @@
+    "free": ["sudo free -m"],
+    "iptables": ["sudo iptables -vnL --line"],
+    "selinux": ["sudo cat /etc/selinux/config"],
+    "os_version": ["sudo cat /etc/issue"],
+    "python_version": ["sudo python -V"],
+    "ram_stats": ["sudo cat /proc/meminfo"],
+    "cron_jobs": ["sudo crontab -l"],
+    "wd_list": ["sudo ls -l ."],
+    "internet_connection": ["sudo curl -D - http://google.com"],
+    "sudoers_list": ["sudo cat /etc/sudoers"],
+    "rotation_configuration": ["sudo cat /etc/logrotate.conf"],
+    "rsyslog_conf": ["sudo cat /etc/rsyslog.conf"],
+    "rsyslog_dir": ["sudo ls -l /etc/rsyslog.d/"],
+    "rsyslog_dir_content": ["sudo find /etc/rsyslog.d/ -type f -exec cat {} \;"],
+    "syslog_ng_conf": ["sudo cat /etc/syslog-ng/syslog-ng.conf"],
+    "syslog_ng_dir": ["sudo ls -l /etc/syslog-ng/conf.d/"],
+    "syslog_ng_dir_content": ["find /etc/syslog-ng/conf.d/ -type f -exec cat {} \;"],
+    "agent_log_snip": ["sudo tail -15 /var/opt/microsoft/omsagent/log/omsagent.log"],
+    "agent_config_dir": ["sudo ls -lR /etc/opt/microsoft/omsagent/"],
+    "agent_cef_config": ["sudo cat /etc/opt/microsoft/omsagent/conf/omsagent.d/security_events.conf"],
+    "tcpdump": ["sudo timeout 2 tcpdump -A -ni any port 25226 -vv"],
+    "top_processes": ["sudo top -bcn1 -w512", "head -n 20"],
+    "omsagent_process": ["sudo ps -aux", "grep omsagent"]
+}
+
+
+def print_notice(input_str):
+    '''
+    Print given text in white background
+    :param input_str: the string whose format will be changed.
+    '''
+    print("\033[0;30;47m" + input_str + "\033[0m")
+
+
+def append_content_to_file(command_object, file_path=output_file_path):
+    """
+
+    :param command_object: consists of the name and the output
+    :param file_path: a file to share the commands outputs
+    """
+    output = repr(command_object)
+    cef_get_info_file = open(file_path, 'a')
+    try:
+        cef_get_info_file.write(output)
+    except Exception:
+        print(str(command_object.command) + "was not documented successfully")
+    cef_get_info_file.close()
+
+
+def run_command(command):
+    """
+
+    :param command: the key value pair from the command dict
+    :return: a command object consisting of a name and an output
+    """
+    if len(command_dict[command]) == 1:
+        command_to_run = subprocess.Popen(command_dict[command][0].split(' '), stdout=subprocess.PIPE,
+                                          stderr=subprocess.STDOUT)
+    # for commands that consist of a single pipe
+    else:
+        first_command = subprocess.Popen(command_dict[command][0].split(' '), stdout=subprocess.PIPE,
+                                         stderr=subprocess.STDOUT)
+        command_to_run = subprocess.Popen(command_dict[command][1].split(' '), stdin=first_command.stdout,
+                                          stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    try:
+        o, e = command_to_run.communicate()
+    except Exception:
+        pass
```

Comment:
We don't want exceptions output?

Reply:
Nope. I don't want customers to deal with exception outputting from with this script. If a command output comes with a fault or empty output that's OK and we can investigate it on our side. The script should just finish running successfully and we will get whatever output is possible to the file.